node/src/node_configurator/mod.rs

```diff
@@ -11,34 +11,218 @@ use crate::database::db_initializer::{DbInitializer, DbInitializerReal};
 use crate::db_config::persistent_configuration::{
     PersistentConfiguration, PersistentConfigurationReal,
 };
-use crate::sub_lib::utils::{db_connection_launch_panic, make_new_multi_config};
+use crate::sub_lib::utils::db_connection_launch_panic;
 use clap::{value_t, App};
+use core::option::Option;
 use dirs::{data_local_dir, home_dir};
 use masq_lib::blockchains::chains::Chain;
 use masq_lib::constants::DEFAULT_CHAIN;
 use masq_lib::multi_config::{merge, CommandLineVcl, EnvironmentVcl, MultiConfig, VclArg};
-use masq_lib::shared_schema::{
-    chain_arg, config_file_arg, data_directory_arg, real_user_arg, ConfiguratorError,
-    DATA_DIRECTORY_HELP,
-};
+use masq_lib::shared_schema::ConfiguratorError;
 use masq_lib::utils::{add_masq_and_chain_directories, localhost};
+use std::env::current_dir;
 use std::net::{SocketAddr, TcpListener};
+use std::ops::Deref;
 use std::path::{Path, PathBuf};
 
 pub trait NodeConfigurator<T> {
     fn configure(&self, multi_config: &MultiConfig) -> Result<T, ConfiguratorError>;
 }
 
-pub fn determine_fundamentals(
+#[derive(Debug)]
+pub struct UserSpecifiedData {
+    pub(crate) chain: Chain,
+    pub(crate) chain_spec: bool,
```

Comment:
These field pairs seem to be duplicated a lot. Should there be a two-field type here instead?

Reply:
```
struct<T> FieldPair {
    item: T,
    user_specified: bool
}
```

Reply:
```
pub struct UserSpecifiedData {
    pub(crate) chain: FieldPair<Chain>,
    pub(crate) real_user: FieldPair<RealUser>,
    // ...
```